### PR TITLE
[Query Manual] - Query custom for books

### PR DESCRIPTION
**Pull request by AleFalcon**
## Summary
Create query custom for books, for filter this.

## Trello Card
https://trello.com/c/U9IiggGo/30-queries-custom-en-repositorio

### CODE REVIEW

File: src/main/java/wolox/training/repositories/BookRepository.java [hunk 1]
```diff
@@ -2,16 +2,20 @@
...
-    //Optional<Book> findByAuthor(String author);
-    //Optional<Book> findOneByAuthor(String author);
     Optional<Book> findByIsbn(String isbn);
     List<Book> findByPublisher(String editor);
-    List<Book> findByGenre(String gender);
+    List<Book> findByGenre(String genre);
     List<Book> findByYear(String year);
+
+    @Query(value = "SELECT * FROM BOOKS B WHERE B.PUBLISHER = ?1 OR B.GENRE = ?2 OR B.YEAR = ?3",
+        nativeQuery = true)
```
Comment (facucompe) [inline src/main/java/wolox/training/repositories/BookRepository.java]: Try not using native query

File: src/main/java/wolox/training/repositories/BookRepository.java [hunk 2]
```diff
@@ -2,16 +2,20 @@
...
 public interface BookRepository extends Repository<Book, Long> {
-    //Optional<Book> findByAuthor(String author);
-    //Optional<Book> findOneByAuthor(String author);
     Optional<Book> findByIsbn(String isbn);
     List<Book> findByPublisher(String editor);
-    List<Book> findByGenre(String gender);
+    List<Book> findByGenre(String genre);
     List<Book> findByYear(String year);
+
+    @Query(value = "SELECT * FROM BOOKS B WHERE B.PUBLISHER = ?1 OR B.GENRE = ?2 OR B.YEAR = ?3",
```
Comment (facucompe) [inline src/main/java/wolox/training/repositories/BookRepository.java]: What if any of this parameters are null

Comment (facucompe) [reply src/main/java/wolox/training/repositories/BookRepository.java]: You can do 
```
SELECT * FROM books
WHERE 
(:publisher is null OR publisher = :publisher) AND
(:genre is null OR genre = :genre) AND ...
```

File: src/main/java/wolox/training/repositories/BookRepository.java [hunk 3]
```diff
@@ -2,16 +2,20 @@
...
+        nativeQuery = true)
+    List<Book> findAllBooksWithFilter(String editor, String genre, String year);
```
Comment (facucompe) [inline src/main/java/wolox/training/repositories/BookRepository.java]: Use `@Param("editor")` annotation so you can use `:editor` instead of `?1`

**Review by facucompe — CHANGES_REQUESTED** [review]

**Review by AleFalcon — COMMENTED** [review]
Ready

**Review by Gianni-Nicolas — APPROVED** [review]

**Review by facucompe — APPROVED** [review]

**Pull request merged**